src/KafkaFlow.OpenTelemetry/OpenTelemetryProducerEventsHandler.cs

```diff
@@ -0,0 +1,109 @@
+                var activity = KafkaFlowActivitySourceHelper.ActivitySource.StartActivity(activityName, ActivityKind.Producer);
+
+                context.Items.Add(KafkaFlowActivitySourceHelper.ActivityString, activity);
+
+                // Depending on Sampling (and whether a listener is registered or not), the
+                // activity above may not be created.
+                // If it is created, then propagate its context.
+                // If it is not created, the propagate the Current context, if any.
+                ActivityContext contextToInject = default;
+
+                if (activity != null)
+                {
+                    contextToInject = activity.Context;
+                }
+                else if (Activity.Current != null)
+                {
+                    contextToInject = Activity.Current.Context;
+                }
+
+                Baggage.Current = Baggage.Create(activity.Baggage.ToDictionary(item => item.Key, item => item.Value));
+
+                // Inject the ActivityContext into the message headers to propagate trace context to the receiving service.
+                Propagator.Inject(new PropagationContext(contextToInject, Baggage.Current), context, InjectTraceContextIntoBasicProperties);
+
+                KafkaFlowActivitySourceHelper.SetGenericTags(activity);
+
+                if (activity != null && activity.IsAllDataRequested)
+                {
+                    SetProducerTags(context, activity);
+                }
+            }
+            catch
+            {
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public static Task OnProducerCompleted(IMessageContext context)
+        {
+            if (context.Items.TryGetValue(KafkaFlowActivitySourceHelper.ActivityString, out var value) && value is Activity activity)
+            {
+                activity.Dispose();
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public static Task OnProducerError(IMessageContext context, Exception ex)
+        {
+            if (context.Items.TryGetValue(KafkaFlowActivitySourceHelper.ActivityString, out var value) && value is Activity activity)
+            {
+                var exceptionEvent = KafkaFlowActivitySourceHelper.CreateExceptionEvent(ex);
+
+                activity?.AddEvent(exceptionEvent);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private static void InjectTraceContextIntoBasicProperties(IMessageContext context, string key, string value)
+        {
+            try
+            {
+                if (!context.Headers.Any(x => x.Key == key))
+                {
+                    context.Headers.SetString(key, value, Encoding.ASCII);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"{ex}. Failed to inject trace context.");
```

Comment:
Should this be removed? Or replaced by a logHandler.

Reply:
The try/catch we included for debugging purposes only. I will removed it

Reply:
Removed here
https://github.com/Farfetch/kafkaflow/pull/451/commits/f02fe4cbaa15aade949cea7ab3b2199cbd831063#diff-e1a8993ccad65fa4162e033fdb038cca41cee39c961538d678241a81e9925cdcL88